activesupport/lib/active_support/i18n_railtie.rb

```diff
@@ -55,7 +55,7 @@ def self.initialize_i18n(app)
 
       reloader = ActiveSupport::FileUpdateChecker.new(I18n.load_path.dup){ I18n.reload! }
       app.reloaders << reloader
-      ActionDispatch::Reloader.to_prepare { reloader.execute_if_updated }
+      ActionDispatch::Reloader.to_prepare { reloader.execute_if_updated.presence }
```

Comment:
Yes, I think the explicit terminator is a better direction. I'm pretty sure we are expecting prepare callbacks to stop if returning false.